cardano-config/src/Cardano/Config/TraceConfig.hs

```diff
@@ -0,0 +1,180 @@
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+{-# OPTIONS_GHC -Wno-partial-fields #-}
+
+module Cardano.Config.TraceConfig
+  (
+    TraceConfig
+  , traceConfigEnabled
+  , traceConfigVerbosity
+  , traceConfigMute
+  , traceConfigParser
+
+  , TraceId
+  , traceEnabled
+
+  -- Per-trace toggles, alpha-sorted.
+  -- Note, the TraceConfig' type itself MUST not be exported.
+  -- The explanation is provided at its declaration site.
+  , traceAcceptPolicy
+  , traceBlockFetchClient
+  , traceBlockFetchDecisions
+  , traceBlockFetchProtocol
+  , traceBlockFetchProtocolSerialised
+  , traceBlockFetchServer
+  , traceChainDB
+  , traceChainSyncClient
+  , traceChainSyncBlockServer
+  , traceChainSyncHeaderServer
+  , traceChainSyncProtocol
+  , traceDnsResolver
+  , traceDnsSubscription
+  , traceErrorPolicy
+  , traceForge
+  , traceHandshake
+  , traceIpSubscription
+  , traceLocalChainSyncProtocol
+  , traceLocalErrorPolicy
+  , traceLocalHandshake
+  , traceLocalTxSubmissionProtocol
+  , traceLocalTxSubmissionServer
+  , traceLocalStateQueryProtocol
+  , traceMempool
+  , traceMux
+  , traceTxInbound
+  , traceTxOutbound
+  , traceTxSubmissionProtocol
+  )
+where
+
+import           Prelude (Show(..))
+import           Cardano.Prelude hiding (show)
+
+import           Data.Aeson
+import           Data.Aeson.Types (Parser)
+
+import           Cardano.BM.Data.Tracer (TracingVerbosity (..))
+
+import           Cardano.Config.Orphanage ()
+
+
+newtype TraceConfig = TraceConfig TraceConfig'
```

Comment:
I think you can remove `newtype TraceConfig = TraceConfig TraceConfig'`, remove it's `Show` instance and redefine `type TraceId = TraceConfig' -> Bool` as `type TraceId = TraceConfig -> Bool`.

Reply:
@Jimbo4350, that `newtype` is necessary to prevent the partiality from escaping from the module -- please, see the explanation in the added comments.

So no, I don't think we can remove it and make the dependent changes you propose.